osu.Game/Screens/Select/SongSelect.cs

```diff
@@ -653,12 +654,28 @@ protected override bool OnKeyDown(KeyDownEvent e)
                         return true;
                     }
 
+                    break;
+                case Key.ShiftLeft:
+                case Key.ShiftRight:
+                    randomFooterButton.FadeText("rewind", 120, Easing.InQuad);
```

Comment:
This should probably be moved into a class like `RandomFooterButton`. Will require a bit of refactoring to the button creation process. I'd probably have it expose a second action (`RewindAction`?) which you could bind the second action to, then all input handling for the random button could be moved inside that class.